Interop/Tensorflow/src/main/java/org/tribuo/interop/tensorflow/TensorFlowModel.java

```diff
@@ -0,0 +1,227 @@
+ */
+public abstract class TensorFlowModel<T extends Output<T>> extends Model<T> implements AutoCloseable {
+    private static final Logger logger = Logger.getLogger(TensorFlowNativeModel.class.getName());
+    private static final long serialVersionUID = 200L;
+
+    protected int batchSize;
+    protected final String initName;
+    protected final String outputName;
+    protected final FeatureConverter<T> featureConverter;
+    protected final OutputConverter<T> outputConverter;
+    protected transient Graph modelGraph = null;
+    protected transient Session session = null;
+    protected transient boolean closed = false;
+
+    /**
+     * Builds a TFModel. The session should be initialized in the subclass constructor.
+     * @param name The model name.
+     * @param provenance The model provenance.
+     * @param featureIDMap The feature domain.
+     * @param outputIDInfo The output domain.
+     * @param trainedGraphDef The graph definition.
+     * @param batchSize The test time batch size.
+     * @param initName The name of the initialization operation.
+     * @param outputName The name of the output operation.
+     * @param featureConverter The feature converter.
+     * @param outputConverter The output converter.
+     */
+    protected TensorFlowModel(String name, ModelProvenance provenance, ImmutableFeatureMap featureIDMap, ImmutableOutputInfo<T> outputIDInfo, GraphDef trainedGraphDef, int batchSize, String initName, String outputName, FeatureConverter<T> featureConverter, OutputConverter<T> outputConverter) {
+        super(name, provenance, featureIDMap, outputIDInfo, outputConverter.generatesProbabilities());
+        this.modelGraph = new Graph();
+        this.modelGraph.importGraphDef(trainedGraphDef);
+        this.session = new Session(modelGraph);
+        this.batchSize = batchSize;
+        this.initName = initName;
+        this.outputName = outputName;
+        this.featureConverter = featureConverter;
+        this.outputConverter = outputConverter;
+    }
+
+    @Override
+    public Prediction<T> predict(Example<T> example) {
+        if (closed) {
+            throw new IllegalStateException("Can't use a closed model, the state has gone.");
+        }
+        // This adds overhead and triggers lookups for each feature, but is necessary to correctly calculate
+        // the number of features used in this example.
+        SparseVector vec = SparseVector.createSparseVector(example, featureIDMap, false);
+        try (TensorMap transformedInput = featureConverter.convert(vec);
+             Tensor outputTensor = transformedInput.feedInto(session.runner())
+                     .fetch(outputName).run().get(0)) {
+            // Transform the returned tensor into a Prediction.
+            return outputConverter.convertToPrediction(outputTensor, outputIDInfo, vec.numActiveElements(), example);
+        }
+    }
+
+    @Override
+    protected List<Prediction<T>> innerPredict(Iterable<Example<T>> examples) {
```

Comment:
Does this method not need to check that the model isn't closed like the other ones do?